coderd/httpapi/name.go

```diff
@@ -79,3 +79,16 @@ func TemplateDisplayNameValid(str string) error {
 	}
 	return nil
 }
+
+// UserRealNameValid returns whether the input string is a valid real user name.
+func UserRealNameValid(str string) error {
+	if len(str) > 64 {
+		return xerrors.New("must be <= 64 characters")
+	}
+
+	trimmed := strings.TrimSpace(str)
+	if trimmed != str {
+		return xerrors.New("must not have leading or trailing white spaces")
+	}
```

Comment:
Let's just store trimmed instead.

Reply:
The idea is to check whether the string got trimmed, which means that there was a whitespace. If I overwrite it, I won't be able to check it.

Alternatively, I can do:

```go
if strings.TrimSpace(str) != str {
```

Reply:
Ah that's fair. I would still just pre-trim the string before validation tbh.

Reply:
Ok, we use `httpapi.Read` to unmarshal and validate the entity. Where would you like to pre-trim it? Before calling `UserRealNameValid`?

Reply:
That's probably a good place to do it. We should probably also trim in the UI/CLI, but I want to avoid a situation where we don't trim ever and you get yelled at if you put a space at the end of your name in the UI.

Reply:
Forgot that the field validator can't modify what it validates. We'll just have to make usre to trim before sending the request.